rust/gui-client/src-tauri/src/client/gui.rs

```diff
@@ -726,6 +740,26 @@ impl Controller {
         }
     }
 
+    async fn update_disabled_resources(&mut self) -> Result<()> {
+        settings::save(&self.advanced_settings).await?;
+
+        let Status::TunnelReady { resources } = &self.status else {
+            bail!("Tunnel is not ready");
+        };
+        let disabled_resources = resources
+            .iter()
+            .filter_map(|r| r.can_toggle().then_some(r.id()))
```

Comment:
I don't think we should guard this here. Instead, connlib should ignore any request to disable resources that can't be disabled.

Reply:
That will make sure this applies to all platforms.

Reply:
(Moving the comment here because it makes more sense)

It involves a bit of duplication but I think only the UI should be concerned about a resource that can be disabled or not, since it's the component that will know what to do with it.

I was designing this trying to have connlib not access the `can_toggle` property.